File: _solutions/07_models/07_models.py
# %%
from chatlas import ChatAuto, ChatPosit

# %% [markdown]
# List models by calling the `list_models` method on a `Chat` instance.

# %%
ChatPosit().list_models()

# %% [markdown]
# You can also load the models into a Polars DataFrame for easier viewing.

# %%
import polars as pl

models = ChatPosit().list_models()
models = pl.DataFrame(models)
models

# %% [markdown]
# Now try sending the same prompt to different models to compare the responses.

# %%
prompt = "Write a recipe for an easy weeknight dinner."

ChatPosit().chat(prompt)
ChatPosit(model="zai-org/GLM-5.3").chat(prompt)

# %%
# If you have local models installed, try them out with Ollama.
ChatAuto("ollama/gemma3:4b").chat(prompt)

File: _solutions/07_models/07_models.R
library(ellmer)

# List models using the `models_posit()` function.
# Hint: try using the Positron data viewer by calling `View()` on the results.
models_posit()

prompt <- "Write a  recipe for an easy weeknight dinner my kids would like."

# Try sending the same prompt to different models to compare the responses.
chat_posit()$chat(prompt)
chat_posit(model = "zai-org/GLM-5.3")$chat(prompt)

# If you have local models installed, you can use them too.
chat_ollama(model = "gemma3:4b")$chat(prompt)

# At home, you'd typically connect to providers directly with API keys, using
# the direct provider functions instead of `chat_posit()`:
#
# chat_openai(model = "gpt-5")
# chat_anthropic(model = "claude-3-7-sonnet-20250219")
